src/core_functions/scalar/list/list_sort.cpp

```diff
@@ -224,8 +228,23 @@ static void ListSortFunction(DataChunk &args, ExpressionState &state, Vector &re
 		}
 
 		D_ASSERT(sel_sorted_idx == incr_payload_count);
-		child_vector.Slice(sel_sorted, sel_sorted_idx);
-		child_vector.Flatten(sel_sorted_idx);
+		if (is_grade_up) {
+			auto &result_entry = ListVector::GetEntry(result);
+			auto result_data = ListVector::GetData(result);
+			auto list_data = ListVector::GetData(input_lists);
+			for (idx_t i = 0; i < count; i++) {
+				result_data[i].length = list_data[i].length;
+				result_data[i].offset = list_data[i].offset;
+				for (idx_t j = result_data[i].offset; j < result_data[i].offset + result_data[i].length; j++) {
+					auto b = sel_sorted.get_index(j) - result_data[i].offset;
+					result_entry.SetValue(j, Value::UINTEGER(b));
```

Comment:
Also, as far as I can tell, you must set the child vector size before writing memory/ setting these values.

I can crash duckdb with this test, which must run.

```sql
statement ok
PRAGMA default_order='ASC';

statement ok
PRAGMA default_null_order='NULLS FIRST';

statement ok
CREATE TABLE tbl AS SELECT list_grade_up(LIST(range)) AS l FROM range(4000);

query I
SELECT l[3000] FROM tbl;
----
3000
```

See `src/include/duckdb/common/types/vector.hpp` for list vector functionality.
```cpp
	//! Sets the total size of the underlying child-vector of a list
	DUCKDB_API static void SetListSize(Vector &vec, idx_t size);
	//! Gets the total capacity of the underlying child-vector of a list
	DUCKDB_API static idx_t GetListCapacity(const Vector &vector);
	//! Sets the total capacity of the underlying child-vector of a list
	DUCKDB_API static void Reserve(Vector &vec, idx_t required_capacity);
```